cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LevelController.java

```diff
@@ -42,26 +44,59 @@ public final class LevelController {
     private static final String LOCATION_LEVEL_ENDPOINT = "levels";
 
-    public Set<SpecifiedLevel> retrieveSpecifiedLevels(ApiConnectionInfo apiConnectionInfo, SpecifiedLevelEndpointInput endpointInput)
+    public Set<SpecifiedLevel> retrieveSpecifiedLevels(ApiConnectionInfo apiConnectionInfo, SpecifiedLevelEndpointInput input)
         throws IOException {
         HttpRequestExecutor executor = new HttpRequestBuilderImpl(apiConnectionInfo, SPECIFIED_LEVEL_ENDPOINT)
-            .addEndpointInput(endpointInput)
+            .addEndpointInput(input)
             .get()
             .withMediaType(ACCEPT_HEADER_V2);
         try (HttpRequestResponse response = executor.execute()) {
             return RadarObjectMapper.mapJsonToSetOfObjects(response.getBody(), SpecifiedLevel.class);
         }
     }
 
-    public LocationLevels retrieveLocationLevels(ApiConnectionInfo apiConnectionInfo, LocationLevelEndpointInput endpointInput)
+    public LocationLevel retrieveLocationLevel(ApiConnectionInfo apiConnectionInfo, LocationLevelEndpointInput.GetOne input)
+        throws IOException {
+        String endpoint = LOCATION_LEVEL_ENDPOINT + "/" + input.levelId();
+        HttpRequestExecutor executor = new HttpRequestBuilderImpl(apiConnectionInfo, endpoint)
+            .addEndpointInput(input)
+            .get()
+            .withMediaType(ACCEPT_HEADER_V2);
+        try (HttpRequestResponse response = executor.execute()) {
+            return RadarObjectMapper.mapJsonToObject(response.getBody(), LocationLevel.class);
+        }
+    }
+
+    public LocationLevels retrieveLocationLevels(ApiConnectionInfo apiConnectionInfo, LocationLevelEndpointInput.GetAll input)
         throws IOException {
         HttpRequestExecutor executor = new HttpRequestBuilderImpl(apiConnectionInfo, LOCATION_LEVEL_ENDPOINT)
-            .addEndpointInput(endpointInput)
+            .addEndpointInput(input)
             .get()
             .withMediaType(ACCEPT_HEADER_V2);
         try (HttpRequestResponse response = executor.execute()) {
             return RadarObjectMapper.mapJsonToObject(response.getBody(), LocationLevels.class);
         }
     }
 
+    public void storeLevel(ApiConnectionInfo apiConnectionInfo, LocationLevelEndpointInput.Post input) throws IOException {
+        String body = RadarObjectMapper.mapObjectToJson(input.level());
+        new HttpRequestBuilderImpl(apiConnectionInfo, LOCATION_LEVEL_ENDPOINT)
+            .addEndpointInput(input)
+            .post()
+            .withBody(body)
+            .withMediaType(ACCEPT_HEADER_V1)
+            .execute()
+            .close();
+    }
+
+    public void deleteLevel(ApiConnectionInfo apiConnectionInfo, LocationLevelEndpointInput.Delete input) throws IOException {
+        String endpoint = LOCATION_LEVEL_ENDPOINT + "/" + input.levelId();
+        new HttpRequestBuilderImpl(apiConnectionInfo, endpoint)
+            .addEndpointInput(input)
+            .delete()
```

Comment:
this is that usage of the delete from before?  So there isn't any state to pass along.